x/axelarcork/keeper/packet.go

```diff
@@ -0,0 +1,92 @@
+package keeper
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	ibcexported "github.com/cosmos/ibc-go/v3/modules/core/exported"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/peggyjv/sommelier/v6/x/axelarcork/types"
+)
+
+// SendPacket wraps IBC ChannelKeeper's SendPacket function
+// If the packet does not get blocked by validation, it passes to the IBC Channel keeper
+func (k Keeper) SendPacket(ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI) error {
+	if err := k.ValidateAxelarCorkPacket(ctx, packet); err != nil {
+		k.Logger(ctx).Error(fmt.Sprintf("ICS20 packet send was denied: %s", err.Error()))
+		return err
+	}
+	return k.ics4Wrapper.SendPacket(ctx, chanCap, packet)
+}
+
+func (k Keeper) ValidateAxelarCorkPacket(ctx sdk.Context, packet ibcexported.PacketI) error {
+	if !k.GetParamSet(ctx).Enabled {
+		return nil
+	}
+
+	// check if this is a call to axelar, exit early if this isn't axelar
+	channelID := packet.GetDestChannel()
+	if channelID != k.GetParamSet(ctx).IbcChannel {
+		return nil
+	}
+
+	// Parse the amount and denom from the packet
+	var packetData transfertypes.FungibleTokenPacketData
+	packetDataBz := packet.GetData()
+	if err := json.Unmarshal(packetDataBz, &packetData); err != nil {
+		return err
+	}
+
+	// if we are not sending to the axelar gmp management account, we can skip
+	if packetData.Receiver != k.GetParamSet(ctx).GmpAccount {
```

Comment:
Are there any other destination accounts we may want to restrict?

Reply:
I assume this is a specific module account, just wondering if there are any other functionalities entailed in other module accounts that we might care about.